supervise.go

```diff
@@ -127,9 +142,35 @@ func (lrb *lineRingBuffer) Lines() []string {
 	return lines
 }
 
+// Stream generates a new channel which will stream any logged lines, including everything currently
+// in the ring buffer. Deregister the stream by calling the close function.
+func (lrb *lineRingBuffer) Stream() (<-chan string, func()) {
+	lrb.Lock()
+	defer lrb.Unlock()
+
+	// Need a chan that has at least len(ring) entries in it, otherwise populating it with existing
```

Comment:
An alternative to sizing the stream channel is to make a copy of the lines and populate the existing contents from a separate goroutine. Then we don’t care how long it takes.

Reply:
Yeah, if we take a copy of lines and then async push them into the channel, its possible other lines can get pushed from `Write` while this is going on (and cause lines to be streamed out of order). Its super unlikely to happen, but something to keep in mind.

I'm skipping this alternative for now, but can go ahead with it if you feel this is a reasonable tradeoff.